### PR TITLE
Merge cbrelease-4.8.9 into 4.8.9.1 #1636

### PR DESCRIPTION
**Pull request by garima468**
Merge cbrelease-4.8.9 into 4.8.9.1 #1636 (User Profile Changes)

### CODE REVIEW

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html [hunk 1]
```diff
@@ -1,49 +1,51 @@
...
-            </ng-container>
+          </ng-container>
         </ng-template>
       </div>
       <div class="block md:hidden">
         <div class="mb-8">
           <div class="flex flex-4 justify-center">
-          <ng-container *ngIf="portalProfile.personalDetails.firstname; else elseFirstNameIcon" >
-            <mat-card-title role="link" tabindex="0" class="flex mobile-margin text-capitalize mat-h2" [innerHTML]="
+            <ng-container *ngIf="portalProfile.personalDetails.firstname; else elseFirstNameIcon">
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html]: optional check should be there

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html [hunk 2]
```diff
@@ -102,22 +110,24 @@
       <div class="flex-1 hidden md:block">
         <div class="flex flex-1">
           <div class="flex flex-4">
-          <ng-container *ngIf="portalProfile.personalDetails.firstname; else elseFirstNameIcon" >
-            <mat-card-title role="link" tabindex="0" class="flex mobile-margin text-capitalize mat-h2" [innerHTML]="
+            <ng-container *ngIf="portalProfile.personalDetails.firstname; else elseFirstNameIcon">
+              <mat-card-title role="link" tabindex="0" class="flex mobile-margin text-capitalize mat-h2" [innerHTML]="
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html:114]: same as above

File: project/ws/app/src/lib/routes/user-profile/components/request-dialog/request-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,119 @@
+<div class="main_container mt-4 flex-row">
+    <div class="">
+        <div class="right_inner_container profile-box">
+            <div class="flex">
+                <h2 class="mat-h1 margin-remove inline-block">
+                    Request to add designation
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/user-profile/components/request-dialog/request-dialog.component.html:6]: I think designation should be Designation

File: project/ws/app/src/lib/routes/user-profile/components/request-dialog/request-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    private requestSvc: RequestService,
+    private dialogRef: MatDialogRef<RequestDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) private data: any
+  ) {
+    this.requestType = this.data.reqType
+    this.userData = this.data
+
+    this.requestForm = new FormGroup({
+      // tslint:disable-next-line:max-line-length
+      designation: new FormControl('', this.requestType === 'Position' ? [Validators.pattern(this.customCharsPattern),
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/user-profile/components/request-dialog/request-dialog.component.ts:61]: Is Position we are using  or coming from api?

File: project/ws/app/src/lib/routes/user-profile/components/request-dialog/request-dialog.component.ts [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+      confirmBox: this.confirm,
+    })
+  }
+
+  submitRequest() {
+    this.disableBtn = true
+    const reqType = this.requestType.toLowerCase()
+    const uniqueID = uuid()
+    this.requestObj = {
+      state: 'INITIATE',
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/user-profile/components/request-dialog/request-dialog.component.ts:81]: hardcoded values should come from constant

File: project/ws/app/src/lib/routes/user-profile/components/request-dialog/request-dialog.component.ts [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+          this.openSnackbar('Your designation request has been successfully submitted')
+         // this.openDialog(this.requestType, _res)
+          this.dialogRef.close()
+        },
+        (err: any) => {
+          this.disableBtn = false
+          if (err.error && err.error.params && err.error.params.errmsg) {
+            this.openSnackbar(err.error.params.errmsg)
+          } else {
+            this.openSnackbar('Something went wrong, please try again later!')
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/user-profile/components/request-dialog/request-dialog.component.ts:116]: same as above

File: project/ws/app/src/lib/routes/user-profile/components/user-profile/user-profile.component.html [hunk 7]
```diff
@@ -413,7 +414,13 @@
                         </mat-form-field>
                       </div>
 
-                      <div class="input-wrapper margin-top-s">
+                      <div class="input-wrapper" *ngIf="!isVerifiedAlready">
+                        <div class="flex flex-1 flex-middle">
+                          <img width="18px" height="18px" mat-raised-button src="assets/icons/info.svg" alt="">
+                          <span class="ml-2">Once you've filled in all the mandatory fields in your profile and receive approval, you will be recognized as a Verified Karmayogi.</span>
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/user-profile/components/user-profile/user-profile.component.html]: Hardcoded vaues should come from constant

File: src/app/routes/public/public-request/request-success-dialog/request-success-dialog.component.ts [hunk 8]
```diff
@@ -52,6 +57,7 @@ export class RequestSuccessDialogComponent implements OnInit {
       }
     } else {
       this.headerMessage = `Your ${this.reqType} request has been successfully submitted`
+      // tslint:disable-next-line: max-line-length
```
Comment (vishnubansaltarento) [inline src/app/routes/public/public-request/request-success-dialog/request-success-dialog.component.ts:60]: same as above

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html (hunk 1 above)
Comment (garima468) [reply project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html]: resolved

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html (hunk 2 above)
Comment (garima468) [reply project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html:114]: resolved

File: project/ws/app/src/lib/routes/user-profile/components/request-dialog/request-dialog.component.html (hunk 3 above)
Comment (garima468) [reply project/ws/app/src/lib/routes/user-profile/components/request-dialog/request-dialog.component.html:6]: It is fine as same coming on the registration page

File: project/ws/app/src/lib/routes/user-profile/components/request-dialog/request-dialog.component.ts (hunk 4 above)
Comment (garima468) [reply project/ws/app/src/lib/routes/user-profile/components/request-dialog/request-dialog.component.ts:61]: It is coming from queryparams

File: project/ws/app/src/lib/routes/user-profile/components/user-profile/user-profile.component.html (hunk 7 above)
Comment (garima468) [reply project/ws/app/src/lib/routes/user-profile/components/user-profile/user-profile.component.html]: resolved

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html [hunk 9]
```diff
@@ -213,18 +229,22 @@
...
               <div class="profile-section">
-                <ws-widget-profile-card-stats [isMobile]="true" (expandCollapse)="expandCollapse($event)" [insightsData]="insightsData" [profileData]="pageData.profileCard" [isLoading]="insightsDataLoading"></ws-widget-profile-card-stats>
+                <ws-widget-profile-card-stats [isMobile]="true" (expandCollapse)="expandCollapse($event)"
+                  [insightsData]="insightsData" [profileData]="pageData.profileCard"
+                  [isLoading]="insightsDataLoading"></ws-widget-profile-card-stats>
               </div>
             </div>
             <div class="margin-bottom-m">
-              <ws-widget-weekly-claps [weeklyData]="pageData.weeklyClaps" [insightsData]="insightsData" [isLoading]="insightsDataLoading"></ws-widget-weekly-claps>
+              <ws-widget-weekly-claps [weeklyData]="pageData.weeklyClaps" [insightsData]="insightsData"
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html:238]: optional check is missing

File: project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html [hunk 10]
```diff
@@ -213,18 +229,22 @@
             <p class="margin-top-s tab-heading">Your stats</p>
             <div class="margin-bottom-m margin-top-m">
               <div class="profile-section">
-                <ws-widget-profile-card-stats [isMobile]="true" (expandCollapse)="expandCollapse($event)" [insightsData]="insightsData" [profileData]="pageData.profileCard" [isLoading]="insightsDataLoading"></ws-widget-profile-card-stats>
+                <ws-widget-profile-card-stats [isMobile]="true" (expandCollapse)="expandCollapse($event)"
+                  [insightsData]="insightsData" [profileData]="pageData.profileCard"
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/profile-v2/routes/profile-view/profile-view.component.html:233]: optional check is missing

**Pull request merged**